src/interactivity/base/ServiceLocator.cpp

```diff
@@ -39,14 +39,30 @@ void ServiceLocator::SetOneCoreTeardownFunction(void (*pfn)()) noexcept
     s_oneCoreTeardownFunction = pfn;
 }
 
-[[noreturn]] void ServiceLocator::RundownAndExit(const HRESULT hr)
+void ServiceLocator::RundownAndExit(const HRESULT hr)

```

Comment:
Changes here are scary. Are we sure this won't cause the OneCore console to regress?

Reply:
No I'm not sure, but I don't believe so. The only effective difference is the addition of the `thread_local` and I could revert the change to `s_shutdownLock` if you'd like.